Client/Frontend/Reader/ReaderMode.swift

```diff
@@ -31,23 +31,16 @@ enum ReaderModeTheme: String {
     case sepia = "sepia"
 
     static func preferredTheme(for theme: ReaderModeTheme? = nil) -> ReaderModeTheme {
-        // If there is no reader theme provided than we default to light theme
-        let readerTheme = theme ?? .light
-        // Get current Firefox theme (Dark vs Normal)
-        // Normal means light theme. This is the overall theme used
-        // by Firefox iOS app
-        let appWideTheme = LegacyThemeManager.instance.currentName
-        // We check for 3 basic themes we have Light / Dark / Sepia
-        // Theme: Dark - app-wide dark overrides all
-        if appWideTheme == .dark {
+        // Get App theme (Dark vs Normal) Normal means light theme.
+        // If appTheme is Dark it will return .dark ReaderModeTheme
+        let themeManager: ThemeManager = AppContainer.shared.resolve()
+        if themeManager.currentTheme.type == .dark {
             return .dark
```

Comment:
We started adding some type specific functions to avoid having to deal with `== .dark`. Maybe we can have one for `getReaderTheme` or something? As an equivalent of `func getThemedImageName(name: String)` for example. Let me know if this make sense or not